pkg/util/knative/knative.go

```diff
@@ -75,7 +75,7 @@ func CreateSubscription(channelReference corev1.ObjectReference, serviceName str
 	}
 }
 
-func CreateTrigger(brokerReference corev1.ObjectReference, serviceName string, eventType string, path string) *eventing.Trigger {
+func CreateTrigger(brokerReference corev1.ObjectReference, serviceName string, serviceAPIVersion string, eventType string, path string) (*eventing.Trigger, error) {
```

Comment:
Altough the logic is correct, I think we're making implicit an implementation detail that should be made explicit making the long term maintenance a possible problem (above all for anybody not knowing this part of the project). I think it's okey to refactor the code and have a private `createTrigger` func like this one, and then two public `CreateTriggerService()` and `CreateTriggerKnativeService` which clarify the intention in the signature. These two func would be calling `createTrigger` with the exepected apiVersion.